services/pushover/service.go

```diff
@@ -0,0 +1,272 @@
+		dec := json.NewDecoder(b)
+		dec.Decode(r)
+		return errors.New(r.Error)
+	}
+
+	return nil
+}
+
+// priority returns the pushover priority as defined by the Pushover API
+// documentation https://pushover.net/api
+func priority(level alert.Level) int {
+	switch level {
+	case alert.OK:
+		// send as -2 to generate no notification/alert
+		return -2
+	case alert.Info:
+		// -1 to always send as a quiet notification
+		return -1
+	case alert.Warning:
+		// 0 to display as high-priority and bypass the user's quiet hours,
+		return 0
+	case alert.Critical:
+		// 1 to also require confirmation from the user
+		return 1
+	}
+
+	return 0
+}
+
+type postData struct {
+	Token    string
+	User     string
+	Message  string
+	Device   string
+	Title    string
+	URL      string
+	URLTitle string
+	Priority int
+	Sound    string
+}
+
+func (p *postData) Values() url.Values {
+	v := url.Values{}
+
+	v.Set("token", p.Token)
+	v.Set("user", p.User)
+	v.Set("message", p.Message)
+	v.Set("priority", strconv.Itoa(p.Priority))
+
+	if p.Device != "" {
+		v.Set("device", p.Device)
+	}
+
+	if p.Title != "" {
+		v.Set("title", p.Title)
+	}
+
+	if p.URL != "" {
+		v.Set("url", p.URL)
+	}
+
+	if p.URLTitle != "" {
+		v.Set("url_title", p.URLTitle)
+	}
+
+	if p.Sound != "" {
+		v.Set("sound", p.Sound)
+	}
+
+	return v
+
+}
+
+func (s *Service) preparePost(user, message, device, title, URL, URLTitle, sound string, level alert.Level) (string, url.Values, error) {
+	c := s.config()
+
+	if !c.Enabled {
+		return "", nil, errors.New("service is not enabled")
+	}
+
+	p := postData{
+		Token:   c.Token,
+		User:    c.User,
+		Message: message,
+	}
+
+	if user != "" {
+		p.User = user
+	}
+
+	p.Device = device
+	p.Title = title
+	p.URL = URL
+	p.URLTitle = URLTitle
+	p.Sound = sound
+
+	p.Priority = priority(level)
+
+	return c.URL, p.Values(), nil
+}
+
+type HandlerConfig struct {
+	// User/Group key of your user (or you), viewable when logged
+	// into the Pushover dashboard. Often referred to as USER_KEY
+	// in the Pushover documentation.
+	// If empty uses the user from the configuration.
+	User string `mapstructure:"user"`
+
+	// Users device name to send message directly to that device,
+	// rather than all of a user's devices (multiple device names may
+	// be separated by a comma)
+	Device string `mapstructure:"device"`
+
+	// Your message's title, otherwise your apps name is used
+	Title string `mapstructure:"title"`
+
+	// A supplementary URL to show with your message
+	URL string `mapstructure:"url"`
+
+	// A title for your supplementary URL, otherwise just URL is shown
+	URLTitle string `mapstructure:"url_title"`
+
+	// The name of one of the sounds supported by the device clients to override
+	// the user's default sound choice
+	Sound string `mapstructure:"sound"`
+
+	// A Unix timestamp of your message's date and time to display to the user,
+	// rather than the time your message is received by the Pushover API
+	Timestamp bool `mapstructure:"timestamp"`
```

Comment:
This is now unused correct? If so we should remove it from here and the pipeline/alert.go PushoverHandler struct as well.